plutus-contract/test/Spec/TxConstraints/MustProduceAtLeast.hs

```diff
@@ -0,0 +1,239 @@
+    ledgerTx2 <- submitTxConstraintsWith @UnitTest lookups2 tx2
+    awaitTxConfirmed $ Tx.getCardanoTxId ledgerTx2
+
+trace :: Contract () Empty ContractError () -> Trace.EmulatorTrace ()
+trace contract = do
+    void $ Trace.activateContractWallet w1 contract
+    void $ Trace.waitNSlots 1
+
+-- Uses onchain and offchain constraint mustProduceAtLeast to spend entire ada balance locked by the script
+entireScriptBalance :: TestTree
+entireScriptBalance =
+    let contract = mustProduceAtLeastContract scriptBalanceValue scriptBalanceValue noTokens $ mockWalletPaymentPubKeyHash w1
+    in  checkPredicateOptions
+            defaultCheckOptions
+            "Successful use of mustProduceAtLeast at script's exact balance"
+            (assertValidatedTransactionCount 2)
+            (void $ trace contract)
+
+-- Uses onchain and offchain constraint mustProduceAtLeast to spend less ada than is locked by the script
+lessThanScriptBalance :: TestTree
+lessThanScriptBalance =
+    let amt = Ada.lovelaceValueOf (scriptBalance -500)
+        contract = mustProduceAtLeastContract amt amt noTokens $ mockWalletPaymentPubKeyHash w1
+    in  checkPredicateOptions
+            defaultCheckOptions
+            "Successful use of mustProduceAtLeast below script's balance"
+            (assertValidatedTransactionCount 2)
+            (void $ trace contract )
+
+-- Uses onchain and offchain constraint mustProduceAtLeast to spend entire token balance locked by the script
+spendTokenBalanceFromScriptAddress :: TestTree
+spendTokenBalanceFromScriptAddress =
+    let token = someTokens 1
+        contract = mustProduceAtLeastContract (scriptBalanceValue <> token) (scriptBalanceValue <> token) token $ mockWalletPaymentPubKeyHash w1
+        options = defaultCheckOptions
+            & changeInitialWalletValue w1 (token <>)
+    in  checkPredicateOptions
+            options
+            "Successful use of mustProduceAtLeast sending tokens"
+            (assertValidatedTransactionCount 2)
+            (void $ trace contract)
+
+-- Uses onchain and offchain constraint mustProduceAtLeast to spend more than the ada balance locked by the script, excess is paid by own wallet.
+higherThanScriptBalance :: TestTree
+higherThanScriptBalance =
+    let amt = Ada.lovelaceValueOf (scriptBalance + 5_000_000)
+        contract = mustProduceAtLeastContract amt amt noTokens $ mockWalletPaymentPubKeyHash w1
+    in  checkPredicateOptions
+            defaultCheckOptions
+            "Validation pass when mustProduceAtLeast is greater than script's balance"
+            (assertValidatedTransactionCount 2)
+            (void $ trace contract)
+
+-- Uses onchain and offchain constraint mustProduceAtLeast to spend more than the ada balance locked by the script. Other wallet's pubkeyhash is provided so cannot pay the excess, asserts contract error.
+  -- FAILS because it successfully gets funds from pubkey, shouldn't ownPaymentPubKeyHash only work for w1, not other wallet?
```

Comment:
No, it should work for any wallet (more specifically PaymentPublicKey) that is provided in the lookups of `ownPaymentPubKeyHash`. In the test, you're providing `w6`'s public key hash in the lookups, so it will use that to cover the excess ADA.